spec/models/miq_report_spec.rb

```diff
@@ -40,6 +40,24 @@
 end
 
 describe MiqReport do
+  context ".for_user" do
+    let(:my_user) { FactoryGirl.create(:user_with_group) }
+    let(:group_in_my_tenant) { FactoryGirl.create(:miq_group, :tenant => my_user.current_tenant) }
+
+    let(:other_tenant) { FactoryGirl.create(:tenant) }
+    let(:group_in_other_tenant) { FactoryGirl.create(:miq_group, :tenant => other_tenant) }
+
+    let!(:my_report) { FactoryGirl.create(:miq_report, :miq_group => my_user.current_group, :rpt_type => "Custom") }
+    let!(:report_in_my_tenant) { FactoryGirl.create(:miq_report, :miq_group => group_in_my_tenant, :rpt_type => "Custom") }
+    let!(:report_in_another_tenant) { FactoryGirl.create(:miq_report, :miq_group => group_in_other_tenant, :rpt_type => "Custom") }
+
+    it "returns reports created by me or anyone in a group in my tenant" do
+      User.current_user = my_user
```

Comment:
Not sure what happened in the specs, but maybe it's not safe to set `User.current_user` as that sets a thread variable?

Reply:
Weird, I had that line there when the tests were passing. I'm gonna restart and hope it succeeds.